clientlib/src/DiscreetORM.ts

```diff
@@ -204,23 +203,27 @@ export function WriteModifiedToDB(discreet_sql_io : DiscreetORMIO){
  * @param discreet_sql_io is the SQL interface.
  */
 function addRow(obj: any, discreet_sql_io : DiscreetORMIO) : void {
-            let add_row_template = "INSERT INTO ? VALUES (?";
-            obj.discreet_orm_id = hash(obj);
-            let obj_hash = obj.discreet_orm_id;
-            let vals_list = [obj.constructor.name,  obj_hash];
-            for (let attribute of Object.keys(obj)){
-                if (attribute === "discreet_orm_id"){
-                    // We want to ignore the secreet discreet_orm_id, since discreet_orm_id is already hardcoded in.
-                    continue;
-                }
-                vals_list.push(obj[attribute]); 
-                add_row_template += ", ?";
-            }
-            add_row_template +=");";
-            let escaped_command = sqlstring.format(add_row_template,vals_list);
-            discreet_sql_io.writeSQL(escaped_command);            
-            console.log(escaped_command);
+    let add_row_template = "INSERT INTO ? VALUES (?";
+    obj.discreet_orm_id = hash(obj);
+    let obj_hash = obj.discreet_orm_id;
+    let vals_list = [obj.constructor.name,  obj_hash];
+    for (let attribute of Object.keys(obj)){
+            // TODO: Object writing is a big feature so we need it in a separate feature
+        if(typeof(obj[attribute]) != "function" && typeof(obj[attribute]) != "undefined" && typeof(obj[attribute]) != "object") {
+            if (attribute === "discreet_orm_id"){
+             // We want to ignore the secreet discreet_orm_id, since discreet_orm_id is already hardcoded in.
+                continue;
+            }
+            vals_list.push(obj[attribute]); 
+            add_row_template += ", ?";
+        }
+    }
+    add_row_template +=");";
+    let escaped_command = sqlstring.format(add_row_template,vals_list);
+    discreet_sql_io.writeSQL(escaped_command);            
+    console.log(escaped_command);
```

Comment:
checks the type of the attribute and prevents it from being added

Reply:
although we'll eventually want to include non-function objects, I think this fix is fine for our current aims